docs/micro-service-architecture.md

```diff
@@ -0,0 +1,248 @@
+# Architecture document
+The goal of this document is to provide a clear overview of our microservice infrastructure, explain our choices and provide a clear OpenAPI definition.
+
+## Technologies
+
+#### `Kubernetes`
+
+Our production architecture leverages Kubernetes for container orchestration with Docker as the containerization platform. 
+This powerful combination provides scalability, flexibility, and efficient management of containerized applications.
+
+#### `Docker`
+
+We mianly use docker for development with many docker-composes to repliate the micro services and their databases but as said before we 
+also use Docker as the containerization platform in our kubernetes environment
+
+#### `NATS`
+
+We chose this solution instead of Kafka for it's lightweight design, low latency and overall simplicity. It also has a built-in mechanism 
+for doing synchronous request-reply messaging, which will be very useful for communication between the API gateway and the microservices. 
+
+#### `Kotlin`
+
+We chose kotlin as our main backend language because we wanted to try something else than java
```

Comment:
```suggestion
We chose Kotlin as our main language for the backend because we wanted to learn a new language that could work in the same environment that Java does.
```